api/src/main/java/org/openmrs/module/pihcore/htmlformentry/action/RegisterBabyAction.java

```diff
@@ -150,6 +161,54 @@ Patient registerBaby(Patient mother, Concept gender, Date birthDatetime, Locatio
         return baby;
     }
 
+    Visit createVisit(Patient baby, Date startDatetime, Encounter motherEncounter) {
+        AdtService adtService = Context.getService(AdtService.class);
+        VisitService visitService = Context.getVisitService();
+        Visit visit = new Visit();
+        visit.setPatient(baby);
+        visit.setLocation(adtService.getLocationThatSupportsVisits(motherEncounter.getLocation()));
+        visit.setStartDatetime(startDatetime);
+        visit.setVisitType(visitService.getVisitTypeByUuid(PihEmrConfigConstants.VISITTYPE_CLINIC_OR_HOSPITAL_VISIT_UUID));
+
+        return visitService.saveVisit(visit);
+    }
+
+    Encounter createAdmissionEncounter(Visit visit, Patient patient, Date birthDatetime, Encounter motherEncounter) {
+        Encounter admissionEncounter = null;
+
+        EncounterService encounterService = Context.getEncounterService();
+        Encounter lastAdmissionEnc = null;
+        List<Encounter> motherEncounters =  motherEncounter.getVisit().getNonVoidedEncounters();
+        // look through mother's encounters for the most recent Admission encounter
+        for (Encounter enc : motherEncounters) {
+            if (enc.getEncounterType().getUuid().equalsIgnoreCase(PihEmrConfigConstants.ENCOUNTERTYPE_ADMISSION_UUID)) {
+                if ( lastAdmissionEnc == null ||
+                        ( (lastAdmissionEnc !=null) && lastAdmissionEnc.getEncounterDatetime().before(enc.getEncounterDatetime()) ) ) {
+                    lastAdmissionEnc = enc;
```

Comment:
We need to take into account transfers within the hospital here as well, so this needs to check transfer encounters as well.

That being said, there's also a utility method from EMR API you should be able to use as well, using a onDate of now, if you wrap the visit in the visit domain wrapper:

https://github.com/openmrs/openmrs-module-emrapi/blob/master/api/src/main/java/org/openmrs/module/emrapi/visit/VisitDomainWrapper.java#L515

Reply:
Thanks @mogoodrich ! I have updated to code to use the Visit Domain Wrapper to retrieve the mother's inpatient location. Thanks!